docs/rfc/block-registration.md

```diff
@@ -69,19 +69,32 @@ To register a new block type, start by creating a `block.json` file. This file:
 	"icon": "star",
 	"description": "Shows warning, error or success notices  ...",
 	"keywords": [ "alert", "message" ],
+	"textDomain": "my-plugin",
 	"attributes": {
 		"message": {
 			"type": "string",
 			"source": "html",
 			"selector": ".meessage"
 		}
 	},
-	"edit": "blocks/notice-edit.js",
-	"save": "blocks/notice-save.js",
 	"styles": [ 
 		{ "name": "default", "label": "Default", "isDefault": true }, 
 		{ "name": "other", "label": "Other" }
-	]
+	],
+	"editorScript": "file:build/editor.assets.json",
+	"script": {
+		"handle": "my-plugin-notice",
+		"file": "index.js",
+		"dependencies": [ "wp-element", "wp-components" ],
+		"version": "3.0.0"
+	},
+	"editorStyle": {
+		"handle": "my-plugin-notice-editor",
+		"file": "editor.css",
+		"dependencies": [ "wp-edit-blocks" ]
+	},
+	"style": "my-plugin-notice",
```

Comment:
What does this mean? I guess handles, for me handles doesn't mean a lot of things outside a WP context. I'd prefer if we avoid them entirely and just use them where absolute necessity (deps)

Shouldn't we just have a single way of defining all these scripts/styles/renderCallback?

Reply:
See the Assets section in RFC which covers it in depth. This tries to bring some backward compatibility. I don't have a strong opinion about it.

Reply:
I’m personally okay with removing the handle from the original proposal and changing `"editorScript": "file:build/editor.asset.json",` to `"editorScript": "file:build/editor.js",` and making the discovery of `asset.json` file an internal thing supported by WordPress given that it might not make sense in the context if only own playground.

Reply:
> Edit: I also figured out that if we don't want to support handles we don't have to use `file:` prefix anymore.

I think it would be nice if we could avoid it, yes.

Depending if we want to support file references for the `icon` property, it's something to consider whether we need to be able to distinguish between an icon slug and and an icon file.

Honestly, I don't think it's much of an issue anyways, particularly if we can either enforce that a file should be defined as relative (will always start with `.`) or has a file extension (will always end in `.` and an extension, e.g. `.svg`).

Reply:
4c53f6a contains the updated proposal.



> > to `"editorScript": "build/editor.js",` and making the discovery of `asset.json` file an internal thing supported by WordPress given that it might not make sense in the context if only own playground.

I took this approach.

> * What if someone wanted to be able to reference an already-registered script or style handle?

They reference the existing file and WordPress ensure they aren't duplicated. I wonder how it is handled as of today.

> * Multiple blocks referencing the same file (like we have today for core blocks)

You can also omit those asset definitions, at least this is what we will initially do for core blocks.

> * Is `file` in `editor.asset.json` even necessary anymore? Or can it be assumed to be its adjacent `.js` file from its own basename?

`file` is not necessary for this scenario.

> * How are `handle` and `version` assigned (notably if this file is auto-generated)?

`handle` can be machine-generated based on the name of the block and asset type:

```js
const handle = kebabCase( `${ block-type }-${ assetType } );
```

`version` can be either generated by webpack using one of its internal hashes or inherited from the plugin's version somehow.